tools/Google.Cloud.Tools.Analyzers/InternalOptionalParametersRequiredAnalyzer.cs

```diff
@@ -106,5 +85,38 @@ private static void AnalyzeInvocation(OperationAnalysisContext context)
                 }
             }
         }
+
+        internal static List<ISymbol> GetVariablesInScope(SyntaxNode syntaxNode, SemanticModel semanticModel) =>
```

Comment:
Yes, I agree. My only point was that if there are multiple omitted arguments, we're going to end up calling `TryGetVariableForArgument` multiple times for the same enumerable returned from `GetVariablesInScope`. Each of those calls will iterate over the same large list of symbols rather than the much smaller filtered list. I think it is better this way, I'm just saying that there's a trade off for a small number of cases.